langkit/gdb/tdh.py

```diff
@@ -47,6 +51,42 @@ def trivia(self, token_no, trivia_no):
                      self._vector_item(self.value['trivias'], trivia_no)['t'],
                      token_no, trivia_no)
 
+    @property  # type: ignore
+    @memoized
+    def _line_starts(self):
+        """
+        Return a python list corresponding to the Lines_Starts vectors in
+        token data handlers. Note that the index of the python list is 0-based
+        whereas the Ada counterpart is 1-based.
+        """
+        lines_starts = self.value['lines_starts']
+        elems = lines_starts['e'].dereference()
+        last = int(lines_starts['size'])
+        return [int(elems[i]) for i in range(1, last)]
+
+    def get_sloc(self, char_index):
+        """
+        Return the Sloc (1-based line and column number) of the character at
+        the given index.
+        """
+        # Get the line number in which this character lies
+        line = int(bisect_right(self._line_starts, char_index)) - 1
+
+        # Get the character index of the first character of the line
+        line_offset = int(self._line_starts[line])
+
+        # Get the last character index of the buffer
+        source_last = int(self.value['source_last'])
+
+        # Compute the column number as being the given character index minus
+        # the character index of the first character of the line. Make sure
+        # the character index is not higher than the index of the last
+        # character of the buffer.
+        column = min(char_index, source_last) - line_offset
```

Comment:
I think it’s ok to have this limitation, but please add a TODO that explains that we don’t support tab expansion here